community_images/common/orchestrator/k8s_setup.py

```diff
@@ -57,48 +57,58 @@ def __enter__(self):
         # bring up the k8s resources
         self.create_k8s_cluster()
 
-        # waiting for pod to be ready
-        logging.info("waiting for pod to be ready")
-        wait_complete = False
-        try:
-            readiness_wait_deployments_suffix = self.runtime_props.get(
-                "readiness_wait_deployments_suffix")
-            if not readiness_wait_deployments_suffix:
-                cmd = f"kubectl wait deployments {self.release_name}"
-                cmd += f" -n {self.namespace_name}"
-                cmd += " --for=condition=Available=True --timeout=20m"
-                Utils.run_cmd(cmd.split())
-            else:
-                for deployment_suffix in readiness_wait_deployments_suffix:
-                    cmd = f"kubectl wait deployments {self.release_name}-{deployment_suffix}"
-                    cmd += f" -n {self.namespace_name}"
-                    cmd += " --for=condition=Available=True --timeout=20m"
-                    Utils.run_cmd(cmd.split())
-
-            wait_complete = True
-        except subprocess.CalledProcessError:
-            logging.info(
-                "Wait deployment command failed, try waiting for pod-0")
+        # if health check script is available, use the health check script to wait
+        # for cluster health
+        if self.runtime_props.get("health_check_script", ""):
+            try:
+                self.run_health_check()
+            except subprocess.CalledProcessError:
+                self.cleanup()
+                raise
 
-        if not wait_complete:
+        else:
+            # waiting for pod to be ready
+            logging.info("waiting for pod to be ready")
+            wait_complete = False
             try:
-                readiness_wait_pod_name_suffix = self.runtime_props.get(
-                    "readiness_wait_pod_name_suffix", ["0"])
-                for pod_name_suffix in readiness_wait_pod_name_suffix:
-                    cmd = ""
-                    if pod_name_suffix == "":
-                        cmd += f"kubectl wait pods {self.release_name}"
-                    else:
-                        cmd += f"kubectl wait pods {self.release_name}-{pod_name_suffix}"
+                readiness_wait_deployments_suffix = self.runtime_props.get(
+                    "readiness_wait_deployments_suffix")
+                if not readiness_wait_deployments_suffix:
+                    cmd = f"kubectl wait deployments {self.release_name}"
                     cmd += f" -n {self.namespace_name}"
-                    cmd += " --for=condition=ready --timeout=20m"
+                    cmd += " --for=condition=Available=True --timeout=20m"
```

Comment:
Please check in k8s if this will work for all cases

Reply:
it was done unintentionally, reverted this change